File: js/problem7.js
function isItPrime(num) {
	var saveTime = Math.ceil(num /2) + 1;
	for (var i = 2; i < saveTime; i++) {
		if (num % i === 0) {
			return false;
		}
	}
	return true;
};

function nth(num, callback) {
	var arr = [];
	for (var i = 2; i < 1000000; i++) {
		if (callback(i)) {
			arr.push(i);
			if (arr.length === num) {
				arr.reverse()
				return arr[0];
			}
		}
	}
};

console.log(nth(10001, isItPrime));

File: python/problem7.py
import math

def isItPrime(n):
	half = int(math.ceil(n // 2) + 1)
	for i in range(2, half):
		if n % i == 0:
			return False
	return True


def nth(num, callback):
	arr = []
	for i in range(2, 1000000):
		if callback(i):
			arr.append(i);
			if len(arr) == num:
				arr.reverse()
				return arr[0]


print(nth(10001, isItPrime))